---
title: "Hi-MC: Overview Guide"
author: "Eric Farber-Eger, Nicholas R. Wheeler, Sandra Smieszek, Dana Crawford"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{1 - Overview Guide}
  %\VignetteEngine{knitr::rmarkdown}
  %\usepackage[utf8]{inputenc}
---
# Hi-MC: High-throughput Mitochondrial haplogroup Classification

### Description

The **Hi-MC** package provides high-level mitochondrial haplogroups given standard PLINK *.map* and *.ped* files.
A reduced phylotree is used as a classification tree to determine samples' haplogroup assignment.

The following is an algorithm description from a recent publication[1] featuring use of the code in this package:

"Phylotree, a comprehensive phylogenetic tree of human mtDNA variation displaying relationships between mitochondrial haplogroups[2], was used as a reference to create a reduced tree of 46 common haplogroups[3]. This reduced classification tree was converted into a node-based tree structure. Each haplogroup node has a list of associated SNPs, a parent node, and zero or more child nodes. The SNPs associated with a node define which SNP genotypes a subject must possess to belong to the corresponding haplogroup. Classification into a haplogroup also requires a subject to recursively meet the definition for the parent haplogroup. Haplogroups that require the reversion to the ancestral genotype (e.g., 10398A-10398G) are accommodated by adding a second hierarchy of required SNP genotypes.

The algorithm determines the appropriate haplogroup in a two-step process. In the first step, the algorithm passes mitochondrial SNP genotype data for each subject into the root node of the tree. The algorithm checks the list of SNP genotypes against those required by the root node. If the genotypes meet the criteria for the parent node, this haplogroup is added to an accumulator. The algorithm then passes the list of SNP genotypes to each of the child nodes connected to that parent node until the tree is exhausted. Next, the algorithm ranks the list of haplogroups in the accumulator according to their distance from the root node. Any haplogroup with a path length less than that of the haplogroup with the longest path length is dropped. The remaining haplogroups, along with their path from the root node to the end node, are returned as a result."[1]

[1] Sandra Smieszek, Sabrina L. Mitchell, Eric H. Farber-Eger, Olivia J. Veatch, Nicholas R. Wheeler, Robert J. Goodloe, Quinn S. Wells, Deborah G. Murdock, Dana C. Crawford (2018) Hi-MC: a novel method for high-throughput mitochondrial haplogroup classification. PeerJ 6:e5149 https://doi.org/10.7717/peerj.5149

[2] Van Oven M, Kayser M. 2009. Updated comprehensive phylogenetic tree of global human mitochondrial DNA variation. Human Mutation 30(2):E386-E394

[3] Mitchell SL, Goodloe R, Brown-Gentry K, Pendergrass SA, Murdock DG, Crawford DC. 2014b. Characterization of mitochondrial haplogroups in a large population-based sample from the United States. Human Genetics 133(7):861-868

The R code in this package was graciously written and provided by Eric Farber-Eger, although all package correspondence should be directed to Dana Crawford (<dana.crawford@case.edu>).

### Installation

```{r, message=FALSE, warning=FALSE, eval=FALSE}
# Install latest package version from CRAN
install.packages("HiMC")
```
**Hi-MC** also relies on the **stringr** and **methods** packages.

### Usage

The **Hi-MC** package takes in standard PLINK *.map* and *.ped* files as arguments.
The *.map* file should have **4** columns and the *.ped* file should have 2**V**+6 columns, where **V** is the number of variants.

Classifying your subjects is a 2-step process.  First, input your *.map* and *.ped* files into **Hi-MC**:
```{r, message=FALSE, warning=FALSE}
# Call library() to load the package
library("HiMC")

# Define file paths to the .map and .ped files
map_file_path <- system.file("extdata", "HapMap_Mito_Seq_QC_2.map", package="HiMC")
ped_file_path <- system.file("extdata", "HapMap_Mito_Seq_QC_2.ped", package="HiMC")

# Use file paths to input the .map and .ped files
snp_data_frame <- HiMC::generate_snp_data(map_file_path,ped_file_path)
```

Then, run the classifier:

```{r, message=FALSE, warning=FALSE}
# Get the HiMC mitochondrial haplogroup classifications
classifications <- HiMC::getClassifications(snp_data_frame)
```

From there, explore the data frame in R, or export it to CSV:

```{r, message=FALSE, warning=FALSE}
# Show the first few rows of selected columns
head(classifications[,c("Individual","haplogroup")])
```
```{r, message=FALSE, warning=FALSE, eval=FALSE}
# Write the dataframe to a .csv file
write.csv(classifications,"classifications.csv",row.names=FALSE)
```

## Caution

The classification tree was built using 54 specifically curated SNPs.

If your dataset is missing more than a few of them, your results will **not** be accurate.

These SNPs referenced are:

+ 825T
+ 1018G
+ 1438A
+ 1719A
+ 1736G
+ 2092T
+ 3505G
+ 3552A
+ 3594C
+ 4580A
+ 4769A
+ 4917G
+ 4977C
+ 5178A
+ 5442C
+ 6371T
+ 7028C
+ 8251A
+ 8414T
+ 8468C
+ 8703T
+ 9042T
+ 9055A
+ 9347G
+ 9950C
+ 10115C
+ 10398G
+ 10398A
+ 10400T
+ 10550G
+ 11177T
+ 11251G
+ 11947G
+ 12007A
+ 12308G
+ 12705C
+ 13263G
+ 13368A
+ 13506C
+ 13708A
+ 13789C
+ 14178C
+ 14318C
+ 14470C
+ 14560A
+ 14668T
+ 14766C
+ 15043A
+ 15326A
+ 15452A
+ 15535T
+ 16111T
+ 16189C
+ 16391A
